src/thing_creature.c

```diff
@@ -2766,6 +2766,110 @@ long project_creature_shot_damage(const struct Thing *thing, ThingModel shot_mod
 
 void creature_fire_shot(struct Thing *firing, struct Thing *target, ThingModel shot_model, char shot_lvl, unsigned char hit_type)
 {
+    if (is_thing_directly_controlled(firing))
+    {
+        if (first_person_dig_claim_mode)
+        {
+            if ( (shot_model == ShM_SwingFist) || (shot_model == ShM_SwingSword) )
+            {
+                if (firing->model == 20)
+                {
+                    if (!(creature_affected_by_spell(firing, SplK_Freeze)))
+                    {
+                        struct Thing* whiptng = NULL;
+                        short stl_x = firing->mappos.x.stl.num;
+                        short stl_y = firing->mappos.y.stl.num;
+                        unsigned char i = 0;
+                        TbBool slappable = false;
```

Comment:
What's all this? Why not link it to hitting with the shot?

Reply:
Because whips are ranged.

Reply:
But units don't carry a literal whip now do they? And there's no animation for said whip. 

So, it's most likely best to take the word 'whipping' not that literal in this context, and take it to mean all the other words of whipping that are still applicable with the action.